frontend/src/components/CustomerDetails.cy.js

```diff
@@ -0,0 +1,88 @@
+import CustomerDetails from '../../src/components/CustomerDetails';
+
+describe('<CustomerDetails />', () => {
+  const customerWithContactAndAddress = {
+    name: 'Acme Corp',
+    employees: 100,
+    size: 'Medium',
+    contactInfo: {
+      name: 'John Doe',
+      email: 'john.doe@example.com',
+    },
+    address: {
+      street: '123 Main St',
+      city: 'Anytown',
+      state: 'CA',
+      zipCode: '12345',
+      country: 'USA',
+    }
+  };
+
+  const customerWithoutContact = {
+    ...customerWithContactAndAddress,
+    contactInfo: null,
+  };
+
+  const customerWithoutAddress = {
+    ...customerWithContactAndAddress,
+    address: null,
+  };
+
+  context('Rendering customer details', () => {
+    it('renders customer details correctly', () => {
+      cy.mount(<CustomerDetails customer={customerWithContactAndAddress} onClick={cy.stub()} />);
+      cy.contains('Customer Details').should('be.visible');
```

Comment:
Deixe uma linha em branco entre ação (ou pré-condição, neste caso) e os resultados esperados.

```suggestion

      cy.contains('Customer Details').should('be.visible');
```